app/src/main/java/org/dosomething/letsdothis/ui/adapters/HubAdapter.java

```diff
@@ -0,0 +1,192 @@
+                return new ProfileViewHolder(profileLayout);
+            case VIEW_TYPE_CURRENT_CAMPAIGN:
+                View currentLayout = LayoutInflater.from(parent.getContext())
+                        .inflate(R.layout.item_hub_current_campaign, parent, false);
+                return new CurrentCampaignViewHolder(currentLayout);
+            case VIEW_TYPE_PAST_CAMPAIGN:
+                View pastLayout = LayoutInflater.from(parent.getContext())
+                        .inflate(R.layout.item_hub_past_campaign, parent, false);
+                return new PastCampaignViewHolder(pastLayout);
+            default:
+                return null;
+        }
+    }
+
+    @Override
+    public void onBindViewHolder(final RecyclerView.ViewHolder holder, final int position)
+    {
+        if(getItemViewType(position) == VIEW_TYPE_SECTION_TITLE)
+        {
+            String s = (String) hubList.get(position);
+            SectionTitleViewHolder sectionTitleViewHolder = (SectionTitleViewHolder) holder;
+            sectionTitleViewHolder.textView.setText(s);
+        }
+        else if(getItemViewType(position) == VIEW_TYPE_PROFILE)
+        {
+            User user = (User) hubList.get(position);
+            ProfileViewHolder profileViewHolder = (ProfileViewHolder) holder;
+            profileViewHolder.fName.setText(user.first_name);
+            profileViewHolder.lName.setText(user.last_name);
+        }
+        else if(getItemViewType(position) == VIEW_TYPE_CURRENT_CAMPAIGN)
+        {
+            Campaign campaign = (Campaign) hubList.get(position);
+            CurrentCampaignViewHolder currentCampaignViewHolder = (CurrentCampaignViewHolder) holder;
+            currentCampaignViewHolder.title.setText(campaign.title);
+            currentCampaignViewHolder.callToAction.setText(campaign.callToAction);
+        }
+        else if(getItemViewType(position) == VIEW_TYPE_PAST_CAMPAIGN)
+        {
+            Campaign campaign = (Campaign) hubList.get(position);
+            PastCampaignViewHolder pastCampaignViewHolder = (PastCampaignViewHolder) holder;
+            //FIXME add image
+            pastCampaignViewHolder.title.setText(campaign.title);
+        }
+    }
+
+    @Override
+    public int getItemViewType(int position)
+    {
+        Object currentObject = hubList.get(position);
+        if(currentObject instanceof String)
+        {
+            return VIEW_TYPE_SECTION_TITLE;
+        }
+        else if(currentObject instanceof User)
+        {
+            return VIEW_TYPE_PROFILE;
+        }
+        else if(currentObject instanceof Campaign)
+        {
+            return VIEW_TYPE_CURRENT_CAMPAIGN;
+        }
+        else if(currentObject instanceof Object)//FIXME detect the past campaign
+        {
+            return VIEW_TYPE_PAST_CAMPAIGN;
+        }
+        return 0;
+    }
+
+    public void addCurrentCampaign(List<Campaign> objects)
+    {
+        hubList.add(1, "currently doing");
```

Comment:
In the case of adding campaigns to current campaigns section, get the index of the past campaign section title. Make them final variables for consistency.